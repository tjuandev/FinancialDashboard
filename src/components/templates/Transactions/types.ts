import { ComponentProps, ReactNode } from 'react'
import { Header } from 'components/organisms'

export type Props = {
  headerProps: ComponentProps<typeof Header>
  children?: ReactNode
}
