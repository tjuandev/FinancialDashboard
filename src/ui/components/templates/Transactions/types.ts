import { ComponentProps } from 'react'
import { Header } from 'ui/components/organisms'
import { Table } from 'ui/components/molecules'

export type TableProps<TableType> = ComponentProps<typeof Table<TableType>>

export type Props<TableType> = {
  headerProps: ComponentProps<typeof Header>
  tableProps: TableProps<TableType>
}
