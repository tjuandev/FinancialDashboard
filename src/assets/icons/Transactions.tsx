type Props = {
  color?: string
}

function Icon({ color }: Props) {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      width="20"
      height="26"
      fill="none"
      viewBox="0 0 20 26"
    >
      <path
        stroke={color ? color : 'currentColor'}
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="3"
        d="M18 24V10.25M10 24V2M2 24v-8.25"
      ></path>
    </svg>
  )
}

export default Icon
