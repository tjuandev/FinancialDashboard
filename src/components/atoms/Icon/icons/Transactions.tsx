import { IconProps } from '../types'

function Icon({ color, height = 26, width = 20 }: IconProps) {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      width={width}
      height={height}
      fill="none"
      role="img"
      aria-label="Transactions"
      viewBox="0 0 20 26"
    >
      <path
        stroke={color ? color : 'currentColor'}
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="3"
        d="M18 24V10.25M10 24V2M2 24v-8.25"
      ></path>
    </svg>
  )
}

export default Icon
