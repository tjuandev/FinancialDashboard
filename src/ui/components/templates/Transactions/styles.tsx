import { Button } from 'ui/components/atoms'
import styled from 'styled-components'
import theme from 'ui/theme'

export const Container = styled.div`
  padding: 0 ${theme.spacing[20]};
`

export const TransactionsSelectorWrapper = styled.div`
  width: 148px;
`

export const ToolbarContainer = styled.div`
  display: flex;
  justify-content: space-between;

  margin-top: 64px;
  margin-bottom: 24px;

  width: 1143px;
`

export const ToolbarLeftSide = styled.div`
  display: flex;
  gap: 1rem;
`

export const ColumnActionsContainer = styled.div`
  display: flex;
  justify-content: end;
  gap: 0.25rem;

  margin: 0.25rem 0;
`

export const NewEntryButton = styled(Button)`
  width: fit-content;
  display: flex;
  align-items: center;
  gap: 10px;

  padding: 12px 16px;

  :hover {
    filter: brightness(0.9);
  }
`

export const Main = styled.main``
