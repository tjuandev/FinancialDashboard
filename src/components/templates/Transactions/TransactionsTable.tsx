import { createColumnHelper } from '@tanstack/react-table'
import { Button, Icon } from 'components/atoms'
import { Table } from 'components/molecules'
import { css } from 'styled-components'
import theme from 'theme'

import { ColumnActionsContainer } from './styles'

type TransactionsTable = {
  type: string
  date: string
  description: {
    value: string
    columnWidth: string
  }
  value: string
}

const columnHelper = createColumnHelper<TransactionsTable>()

const actionButtonExtraStyles = css`
  width: 40px;
  height: 40px;

  border-radius: ${theme.borderRadius.base};

  :hover {
    filter: brightness(0.9);
  }
`

const tableColumns = [
  columnHelper.accessor('type', {
    header: 'Type',
    cell: (info) => info.getValue()
  }),
  columnHelper.accessor('date', {
    header: 'Date',
    cell: (info) => info.getValue()
  }),
  columnHelper.accessor('description', {
    header: 'Description',
    cell: (info) => info.getValue().value
  }),
  columnHelper.accessor('value', {
    header: 'Value',
    cell: (info) => info.getValue()
  }),
  columnHelper.display({
    id: 'actions',
    header: 'Actions',
    cell: () => (
      <ColumnActionsContainer>
        <Button padding="0.25rem" extraStyles={actionButtonExtraStyles}>
          <Icon name="outlined-trash" />
        </Button>
        <Button padding="0.25rem" extraStyles={actionButtonExtraStyles}>
          <Icon name="outlined-edit" />
        </Button>
      </ColumnActionsContainer>
    )
  })
]

const rows: TransactionsTable[] = [
  {
    type: 'Despesa',
    date: '2022-02-2004',
    description: {
      value: 'Compras no submundo pós apocalíptico de Gânador',
      columnWidth: '100px'
    },
    value: '2000'
  },
  {
    type: 'Despesa',
    date: '2022-02-2004',
    description: {
      value: 'Compras no submundo pós apocalíptico de Gânador',
      columnWidth: '100px'
    },
    value: '2000'
  },
  {
    type: 'Despesa',
    date: '2022-02-2004',
    description: {
      value: 'Compras no submundo pós apocalíptico de Gânador',
      columnWidth: '100px'
    },
    value: '2000'
  }
]

const TransactionsTable = () => {
  return (
    <Table<TransactionsTable>
      columns={tableColumns}
      rows={rows}
      columnsExtensions={[
        {
          id: 'actions',
          horizontalAlign: 'end'
        }
      ]}
    />
  )
}

export default TransactionsTable
