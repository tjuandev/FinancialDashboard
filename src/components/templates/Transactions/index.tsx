import * as S from './styles'

import { Grid, Icon } from 'components/atoms'
import { ExpandableInput, Kpi, Select, Table } from 'components/molecules'
import { Header } from 'components/organisms'

import { Props, TableProps } from './types'

import { IconLiterals } from 'types/icons'

import { v4 as uuid } from 'uuid'

const kpisMock = [
  {
    name: 'R$ 1290,20',
    description: 'Seu saldo atual',
    iconName: 'wallet'
  },
  {
    name: 'R$ 2080,63',
    description: 'Seu ganhos',
    iconName: 'stock-up'
  },
  {
    name: 'R$ 790,43',
    description: 'Suas despesas',
    iconName: 'stock-down'
  }
]

const KPIs = () => (
  <Grid columns="repeat(3, 290px)" gap={8}>
    {kpisMock.map(({ iconName, ...kpiProps }) => (
      <Kpi
        headerChildren={<Icon name={iconName as IconLiterals} />}
        key={uuid()}
        {...kpiProps}
      />
    ))}
  </Grid>
)

const TransactionsSelector = () => (
  <S.TransactionsSelectorWrapper>
    <Select
      defaultValue="all"
      options={[
        { value: 'all', label: 'Todas' },
        { value: 'incomes', label: 'Ganhos' },
        { value: 'outcomes', label: 'Despesas' }
      ]}
    />
  </S.TransactionsSelectorWrapper>
)

const Toolbar = () => (
  <S.ToolbarContainer>
    <TransactionsSelector />
    <S.ToolbarLeftSide>
      <ExpandableInput placeholder="Type something..." />
      <S.NewEntryButton colorSchema="secondary">
        Nova Entrada <Icon name="plus" />
      </S.NewEntryButton>
    </S.ToolbarLeftSide>
  </S.ToolbarContainer>
)

const TransactionsTable = <TableType,>(props: TableProps<TableType>) => {
  const { columns, rows } = props
  const shouldLoad = !columns && !rows

  return <Table loading={shouldLoad} {...props} />
}

const Transactions = <TableType,>({
  headerProps,
  tableProps
}: Props<TableType>) => (
  <S.Container>
    <Header {...headerProps} />
    <S.Main>
      <KPIs />
      <Toolbar />
      <TransactionsTable {...tableProps} />
    </S.Main>
  </S.Container>
)

export default Transactions
