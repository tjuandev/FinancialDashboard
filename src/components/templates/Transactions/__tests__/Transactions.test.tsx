/* eslint-disable @typescript-eslint/no-explicit-any */
import { render } from '@testing-library/react'
import Transactions from '..'

jest.mock('uuid')

// NOTE Move it to testHelper after

jest.mock('uuid', () => {
  const base = '9134e286-6f71-427a-bf00-'
  let current = 100000000000

  return {
    v4: () => {
      const uuid = base + current.toString()
      current++

      return uuid
    }
  }
})

describe('<Transactions />', () => {
  it('Should match snapshot', () => {
    const { container } = render(
      <Transactions<any>
        headerProps={{
          avatarProps: {
            name: 'User Test'
          },
          breadcrumbProps: {
            name: 'Test'
          }
        }}
        tableProps={{
          columns: [],
          rows: []
        }}
      />
    )

    expect(container).toMatchSnapshot()
  })
})
