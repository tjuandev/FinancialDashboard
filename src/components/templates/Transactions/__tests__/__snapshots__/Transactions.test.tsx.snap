// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Transactions /> Should match snapshot 1`] = `
.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.5rem;
}

.c8 {
  text-align: right;
}

.c8 p {
  font-weight: 600;
  font-size: 0.875rem;
  line-height: 12px;
}

.c8 small {
  font-size: 0.75rem;
}

.c9 {
  width: 48px;
  height: 48px;
  position: relative;
}

.c9,
.c9 img {
  border-radius: 0.625rem;
}

.c10 {
  width: 100%;
  height: 100%;
  border-radius: inherit;
  background: #CDDBC2;
  font-weight: bolder;
  font-size: 1.25rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c6 {
  height: 44px;
  width: 44px;
  background-color: #CDDBC2;
  color: #14404D;
  padding: 0.5rem;
  border-radius: 0.625rem;
  font-size: 1rem;
  font-weight: 600;
  -webkit-transition: all 0.25s ease;
  transition: all 0.25s ease;
  cursor: pointer;
}

.c6:focus {
  outline: 1px solid #14404D;
}

.c21 {
  height: 44px;
  width: 44px;
  background-color: #14404D;
  color: #CDDBC2;
  padding: 0.5rem;
  border-radius: 0.625rem;
  font-size: 1rem;
  font-weight: 600;
  -webkit-transition: all 0.25s ease;
  transition: all 0.25s ease;
  cursor: pointer;
}

.c21:focus {
  outline: 1px solid #14404D;
}

.c11 {
  display: grid;
  gap: 2rem;
  grid-template-columns: repeat(3,290px);
  width: 100%;
  height: 100%;
}

.c0 {
  padding: 0 5rem;
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-top: 64px;
  margin-bottom: 24px;
  width: 1143px;
}

.c17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 1rem;
}

.c22 {
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 10px;
  padding: 12px 16px;
}

.c22:hover {
  -webkit-filter: brightness(0.9);
  filter: brightness(0.9);
}

.c12 {
  border-radius: 1.25rem;
  padding: 1.5rem;
  width: 100%;
  height: 100%;
  background: #CDDBC2;
  color: #14404D;
}

.c12 > h2 {
  font-size: 1.875rem;
  margin-bottom: 4px;
}

.c12 > p {
  font-size: 1rem;
}

.c13 {
  margin-bottom: 1.5rem;
}

.c15 > div {
  border: none;
  background: #CDDBC2;
  border-radius: 0.625rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0.5rem 1rem;
  cursor: pointer;
}

.c15 > div > div {
  padding: 0;
}

.c16 {
  grid-area: 1/1/2/3;
}

.c16 > div {
  padding: 0;
  margin: 0;
}

.c18 {
  height: 44px;
  width: 44px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  background: #CDDBC2;
  border-radius: 0.625rem;
  -webkit-transition: all 0.25s ease;
  transition: all 0.25s ease;
  cursor: pointer;
  cursor: pointer;
}

.c18.open {
  width: 300px;
}

.c18:not(.open):hover {
  -webkit-filter: brightness(0.8);
  filter: brightness(0.8);
}

.c18:focus-within {
  outline: 0.5px solid #14404D;
}

.c20 {
  position: relative;
  margin: 0 16px 0 40px;
  background: transparent;
  border: 0;
  font-size: 1rem;
  -webkit-transition: all 0.25s ease;
  transition: all 0.25s ease;
  opacity: 0;
  width: 0;
}

.c20:focus {
  outline: 0;
}

.c20.open {
  opacity: 1;
  width: 100%;
}

.c19 > button {
  position: absolute;
  top: 0;
  left: 0;
  border: 0;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.5rem;
}

.c3 > h1 {
  font-size: 2rem;
}

.c4 {
  width: 24px;
  height: 24px;
}

.c23 {
  width: 100%;
  border-collapse: collapse;
}

.c23 thead {
  background: #14404D;
  border-radius: 1rem;
}

.c23 tbody {
  background: #CDDBC2;
}

.c23 th,
.c23 td {
  font-size: 14px;
}

.c23 td {
  padding: 0 16px;
}

.c23 tr:last-child td:first-child {
  border-radius: 0 0 0 16px;
}

.c23 tr:last-child td:last-child {
  border-radius: 0 0 16px 0;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-top: 1.5rem;
  margin-bottom: 4rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 2rem;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 2rem;
}

<div>
  <div
    class="c0"
  >
    <header
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <div
            class="c4"
          />
          <h1>
            Test
          </h1>
        </div>
      </div>
      <div
        class="c5"
      >
        <button
          class="c6"
        >
          <svg
            aria-label="Notification"
            fill="none"
            height="22"
            role="img"
            viewBox="0 0 20 22"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16 7A6 6 0 104 7c0 7-3 9-3 9h18s-3-2-3-9zM11.73 20a1.999 1.999 0 01-3.46 0"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </button>
        <div
          class="c7"
        >
          <div
            class="c8"
          >
            <p>
              User Test
            </p>
            <small />
          </div>
          <div
            class="c9"
          >
            <div
              class="c10"
            >
              UT
            </div>
          </div>
        </div>
      </div>
    </header>
    <main
      class=""
    >
      <div
        class="c11"
      >
        <div
          class="c12"
        >
          <header
            class="c13"
          >
            <svg
              aria-label="Wallet"
              fill="none"
              height="38"
              role="img"
              viewBox="0 0 38 34"
              width="34"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M32.884 9.714H8c-.592 0-1.714-.428-1.714-2.143 0-1.714 1.55-1.714 2.143-1.714h24.455c.592 0 1.259-.694 1.259-1.286 0-1.775-1.44-2.571-3.214-2.571H6.286A4.285 4.285 0 002 6.286v21.428A4.285 4.285 0 006.286 32h26.598c1.876 0 3.402-1.442 3.402-3.214V11.643c0-1.773-1.526-1.929-3.402-1.929zm-3.027 12.643a2.143 2.143 0 110-4.286 2.143 2.143 0 010 4.286z"
                stroke="currentColor"
                stroke-width="3"
              />
            </svg>
          </header>
          <h2>
            R$ 1290,20
          </h2>
          <p>
            Seu saldo atual
          </p>
        </div>
        <div
          class="c12"
        >
          <header
            class="c13"
          >
            <svg
              aria-label="Stock Up"
              fill="none"
              height="29"
              role="img"
              viewBox="0 0 35 29"
              width="35"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                clip-path="url(#clip0_6_486)"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="3"
              >
                <path
                  d="M33.542 7.25L19.687 18.73l-7.291-6.043L1.458 21.75"
                />
                <path
                  d="M24.792 7.25h8.75v7.25"
                />
              </g>
              <defs>
                <clippath
                  id="clip0_6_486"
                >
                  <path
                    d="M0 0H35V29H0z"
                    fill="#fff"
                  />
                </clippath>
              </defs>
            </svg>
          </header>
          <h2>
            R$ 2080,63
          </h2>
          <p>
            Seu ganhos
          </p>
        </div>
        <div
          class="c12"
        >
          <header
            class="c13"
          >
            <svg
              fill="none"
              height="29"
              viewBox="0 0 36 29"
              width="35"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M34.5 21.75L20.25 10.27l-7.5 6.043L1.5 7.25"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="3"
              />
              <path
                d="M25.5 21.75h9V14.5"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="3"
              />
            </svg>
          </header>
          <h2>
            R$ 790,43
          </h2>
          <p>
            Suas despesas
          </p>
        </div>
      </div>
      <div
        class="c14"
      >
        <div
          class=""
        >
          <div
            class=" css-b62m3t-container"
          >
            <span
              class="css-1f43avz-a11yText-A11yText"
              id="react-select-2-live-region"
            />
            <span
              aria-atomic="false"
              aria-live="polite"
              aria-relevant="additions text"
              class="css-1f43avz-a11yText-A11yText"
            />
            <div
              class="c15"
              data-testid="Control"
            >
              <div
                class=" css-1s2u09g-control"
              >
                <div
                  class=" css-319lph-ValueContainer"
                >
                  <div
                    class=" css-qnpei2-placeholder"
                    id="react-select-2-placeholder"
                  >
                    Select...
                  </div>
                  <div
                    class="c16"
                    data-testid="Input"
                  >
                    <div
                      class=" css-1feah6e-Input"
                      data-value=""
                    >
                      <input
                        aria-autocomplete="list"
                        aria-describedby="react-select-2-placeholder"
                        aria-expanded="false"
                        aria-haspopup="true"
                        autocapitalize="none"
                        autocomplete="off"
                        autocorrect="off"
                        class=""
                        id="react-select-2-input"
                        role="combobox"
                        spellcheck="false"
                        style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                        tabindex="0"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </div>
                <div
                  class=" css-1hb7zxy-IndicatorsContainer"
                >
                  <span
                    class=" css-1okebmr-indicatorSeparator"
                  />
                  <svg
                    aria-label="ChevronDown"
                    fill="none"
                    height="9"
                    role="img"
                    viewBox="0 0 15 9"
                    width="15"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M2 2.244l5.5 4.512L13 2.244"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="3"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="c17"
        >
          <div
            aria-expanded="false"
            class="c18"
            data-testid="container-element"
          >
            <div
              class="c19"
            >
              <button
                class="c6"
              >
                <svg
                  fill="none"
                  height="23"
                  viewBox="0 0 23 23"
                  width="23"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M10.444 18.889a8.444 8.444 0 100-16.889 8.444 8.444 0 000 16.889zM21 21l-4.592-4.592"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="3"
                  />
                </svg>
              </button>
            </div>
            <input
              class="c20"
              placeholder="Type something..."
              type="text"
            />
          </div>
          <button
            class="c21 c22"
          >
            Nova Entrada 
            <svg
              fill="none"
              height="16"
              viewBox="0 0 16 17"
              width="17"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 3.833v9.334M3.333 8.5h9.334"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="3"
              />
            </svg>
          </button>
        </div>
      </div>
      <table
        class="c23"
      >
        <thead>
          <tr />
        </thead>
        <tbody />
      </table>
    </main>
  </div>
</div>
`;
